### DISCUSSION TITLE
CloudWatch: AWS/MediaLive namespace is missing on the Metrics selector

### DISCUSSION THREAD

**Discussion by ethirolle** [grafana/grafana#37079] [Plugins]
CloudWatch plugin still uses a hard-coded list of namespaces and metrics, at least until https://github.com/grafana/grafana/issues/30977 is fixed & merged.

A Grafana Cloud customer noted that this **namespace** is not included in the plugin, and they would like us to add it:

```
AWS/MediaLive
```

I asked him for a list of the **metric names** he would like to have under that namespace, and he provided this:

> We need to have at least:
> - ActiveAlerts
> - Network In (Mbps)
> - Network Out (Mbps)
> - InputVideoFrameRate

I also asked him for a place in the AWS MediaLive docs where it lists the metric names that can be monitored by Cloudwatch, and he provided these:

- https://aws.amazon.com/blogs/media/monitoring-aws-media-services-using-amazon-cloudwatch-events/
- https://docs.aws.amazon.com/medialive/latest/ug/requirements-for-monitor-channel-health.html

Thanks!!

![image](https://user-images.githubusercontent.com/10763196/126565288-475bee8b-08fd-4933-9429-116f8f19b4b8.png)

**Comment by ethirolle** [comment]
Note: this is similar to prior CloudWatch feature requests like https://github.com/grafana/grafana/issues/36819.

**Comment by sarahzinger** [comment]
Hi there,

As part of our work to move from [discussions to issues](https://github.com/grafana/grafana/discussions/73424), we're looking to see what lingering discussions we have that we can close up or move to issues. If I understand properly, it seems like the original issue here was resolved, although the larger issue that we're using a hard coded list for metrics remains, unfortunately I think this is a limitation of the aws sdk for now. But perhaps in the future that could change! Otherwise I think the best thing to do is to create issues/prs to update this hardcoded list of metrics when we need it.
